src/elysium/sigmawallet.cpp

```diff
@@ -503,4 +472,17 @@ bool SigmaWallet::RemoveFromMintPool(SigmaPublicKey const &publicKey)
     return false;
 }
 
-} // exodus
+// Helper
+std::unique_ptr<CWalletDB> SigmaWallet::EnsureDBConnection(CWalletDB* &db) const
```

Comment:
This is quite dangerous due to if the caller does not declare a variable to hold the lifetime it will be broken.

Reply:
Nice catch